### DISCUSSION TITLE
Small font icons

### DISCUSSION THREAD

**Discussion by anatskiy** [ghostty-org/ghostty#3501] [Feature Requests, Ideas] [answered]
I'm having issues with font icons that look small compared to the font itself (especially in the gutter area in Neovim). They're fine in, for example, Wezterm. I'm not sure if this is related to #3029.

Ghostty left, Wezterm right:
<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/fbd83f07-6489-4129-9260-0edc5e4b6228" />

Used font: JetBrainsMono Nerd Font
Ghostty version: 1.0.0 build 8469
Hardware: MacBook Pro M1 (if it's relevant)

Ghostty config:
```
font-family = JetBrainsMono Nerd Font Mono
font-size = 15

theme = catppuccin-mocha
background-opacity = 0.85
background-blur-radius = 20
```

**Comment by zuqini** [comment]
Looks like a duplicate of https://github.com/ghostty-org/ghostty/issues/1903

I fixed a similar issue with `Iosevka Nerd Font` by simply switching over to use the unpatched version of the font, which I _think_ addresses the issue by falling back to Ghostty's default symbols font? Someone can correct me if that's not accurate.

**Comment by anatskiy** [comment]
Ah, I only looked in _open_ issues. Yeah, having a non-nerd font as primary and an icons-only font as fallback helped. I'm closing the issue.

**Comment by zuqini** [comment]
@anatskiy In my case I didn't actually need to specify a symbols only fallback font. I believe Ghostty supplies its own so you might be able to get away without it too.

**Comment by anatskiy** [comment]
@zuqini The only font config that actually worked for me is (the first two lines are only for the sake of explicitness):

```
font-family = ""
font-family = JetBrains Mono
font-family = Symbols Nerd Font Mono
font-size = 15
```

Unfortunately, this brings another issue. Icons in some apps like Lazygit and Neovim have improper scaling when selected/hovered:

https://github.com/user-attachments/assets/e0da04b7-9a6c-43b2-a4bd-f38bcad90797

https://github.com/user-attachments/assets/7d7ae34b-c69d-4ece-a7f0-77be9011b02a

This is better than small icons but still very annoying. I'll see how much of a dealbreaker this is for me. In Wezterm and iTerm2 with exactly the same config (main font JetBrains Mono, fallback font Symbols Nerd Font Mono) icons render as expected.

@mitchellh Is this worth looking into?

**Reply by datacurse** [reply]
thanks, that did the job for me!

**Comment by mitchellh** [comment]
Convert to discussion for issue triage.

**Comment by anatskiy** [comment]
Hey @mitchellh, after a lot of trial and error, that's what I found out. Without setting any font settings (except for the font size), icons look okay-ish:

<img width="283" alt="1" src="https://github.com/user-attachments/assets/315a84e5-f349-4bd2-9d55-d31674dd23d4" />

When I explicitly set the font to `JetBrains Mono`, nothing changes. When I set a fallback font in addition to the primary one, icons look bigger (nice), but when the text is bold, the icons become smaller and shift to the left. I assume that's because the fallback font doesn't have a bold variant, and Ghostty uses the regular variant (not Mono).
```
font-family = JetBrains Mono
font-family = Symbols Nerd Font Mono
```

<img width="289" alt="2a" src="https://github.com/user-attachments/assets/2620503c-7ef2-4865-8b6d-707b0560835c" />

<img width="655" alt="2b" src="https://github.com/user-attachments/assets/040835c5-3cc9-43f3-9ad7-62ae948f212d" />

However, when I only set the nerd font (without the primary one), everything looks very nicely, even icons render as bold somehow.
```
font-family = Symbols Nerd Font Mono
```

<img width="338" alt="3" src="https://github.com/user-attachments/assets/5a3726cf-4c5d-4ca9-970a-f1f9eef54df0" />

This is exactly the outcome I wanted and I could call it a day but I would really love to understand what's going on and why setting both `JetBrains Mono` and `Symbols Nerd Font Mono` doesn't work as expected.

**Comment by gcollura** [comment]
For me, not all icons are small. This is my config:
```
command = /bin/zsh
initial-command = /bin/zsh -l -c 'tmux new-session -A -s personal'

auto-update-channel = stable

font-family = "JetBrains Mono"
font-style = "Light"
font-size = 11
adjust-cell-width = -1
theme = catppuccin-frappe
click-repeat-interval = 500

window-padding-x = 0
window-padding-y = 0
window-padding-balance = true
window-save-state = always
```

<img width="442" alt="image" src="https://github.com/user-attachments/assets/4e2b0a0f-5713-4d84-a208-08b30cd91a74" />

(this is oil.nvim in neovim 0.10.3)

The same happens in other extensions, eg. Telescope:
<img width="188" alt="image" src="https://github.com/user-attachments/assets/1ffe773c-2f8c-499d-9418-a75e025713bb" />


In Wezterm, icons all have the same size, as expected:
<img width="462" alt="image" src="https://github.com/user-attachments/assets/f0d10d9b-ccc4-423d-9d03-86998ed550d0" />


I think it might have something to do to how winblend works with neovim, once I disable it, I get:
<img width="351" alt="image" src="https://github.com/user-attachments/assets/012bf70e-5dbf-4a3a-a9aa-a356f29d9ce6" />


Finally, with:
```
font-family = ""
font-family = Symbols Nerd Font Mono

font-style = 
font-size = 11
```

<img width="350" alt="image" src="https://github.com/user-attachments/assets/b056fbd6-768c-45f0-9a7b-cce1e793048a" />

**Comment by 00-kat** [comment, marked as answer]
For the record, the following fixed it for me, stolen from [Kitty's FAQ] and translated:
```ini
font-codepoint-map = U+e000-U+e00a,U+ea60-U+ebeb,U+e0a0-U+e0c8,U+e0ca,U+e0cc-U+e0d7,U+e200-U+e2a9,U+e300-U+e3e3,U+e5fa-U+e6b1,U+e700-U+e7c5,U+ed00-U+efc1,U+f000-U+f2ff,U+f000-U+f2e0,U+f300-U+f372,U+f400-U+f533,U+f0001-U+f1af0=Symbols Nerd Font Mono
```
Note: you need to install the Nerd Font symbols from [their GitHub releases page] for this to work (you're looking for `NerdFontsSymbolsOnly.zip` or `NerdFontsSymbolsOnly.tar.gz`).

Edit: see the comment below this for updated ranges.

[Kitty's FAQ]: https://sw.kovidgoyal.net/kitty/faq/#kitty-is-not-able-to-use-my-favorite-font
[their GitHub releases page]: https://github.com/ryanoasis/nerd-fonts/releases

**Reply by net** [reply]
I updated it to the current Nerd Font ranges:

```ini
font-codepoint-map = U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0C8,U+E0CA,U+E0CC-U+E0D7,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E6B7,U+E700-U+E8EF,U+EA60-U+EC1E,U+ED00-U+F2FF,U+EE00-U+EE0B,U+F300-U+F381,U+F400-U+F533,U+F0001-U+F1AF0=Symbols Nerd Font Mono
```

Also, `Symbols Nerd Font` looked better than `Symbols Nerd Font Mono` to me:

```ini
font-codepoint-map = U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0C8,U+E0CA,U+E0CC-U+E0D7,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E6B7,U+E700-U+E8EF,U+EA60-U+EC1E,U+ED00-U+F2FF,U+EE00-U+EE0B,U+F300-U+F381,U+F400-U+F533,U+F0001-U+F1AF0=Symbols Nerd Font
```

**Reply by nstrelow** [reply]
OMG this is so much nicer now.

I had some problems with some powerline icons being smaller than the rest of the "powerline", e.g. "U+E0B4"

Fixed it by removing this range "U+E0B0-U+E0C8".

Might also be that I need to add other icons the the map 🤷‍♂️

**Comment by esn89** [comment]
No luck with this setting, still:

<img width="462" alt="Screenshot 2025-04-03 at 11 41 06 AM" src="https://github.com/user-attachments/assets/d326319e-2ded-4103-b6bc-6734ebc75b24" />

**Comment by glats** [comment]
for me i checked my ~/.config/fontconfig and i removed i custom file called `10-nerd-font-symbols.conf`
here is the content just in case https://gist.github.com/j1cs/f6db8b210e36ebeb075d6435f005d9fc
